source/Flotsam.cpp

```diff
@@ -67,6 +67,17 @@ void Flotsam::Place(const Ship &source)
 
 
 
+// Place this flotsam with its starting position at the specified bay of the source ship,
+// instead of the center of the ship.
+void Flotsam::Place(const Ship &source, size_t bayIndex)
+{
+	Place(source);
+	if(source.Bays().size() > bayIndex)
+		this->position += source.Bays().at(bayIndex).point;
```

Comment:
You already have a bounds check in the `if`.
```suggestion
		this->position += source.Bays()[bayIndex].point;
```